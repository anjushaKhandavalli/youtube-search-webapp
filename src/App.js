import React from 'react';
import { AppContainer } from './App.style'
import SearchInputBox from './components/SearchInputBox';

const App = () => {
  return (
    <AppContainer>
      <SearchInputBox/>
    </AppContainer>
  );
}

export default App;
