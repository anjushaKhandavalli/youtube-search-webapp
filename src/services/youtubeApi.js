import axios from 'axios';
const KEY = 'Enter your Api Key here';

export default axios.create({
    baseURL: 'https://www.googleapis.com/youtube/v3/',
    params: {
        part: 'snippet',
        maxResults: 10,
        key: KEY
    }
})