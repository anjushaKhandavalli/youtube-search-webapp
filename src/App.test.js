import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import App from './App';
import youtubeApi from './services/youtubeApi';

jest.mock('./services/youtubeApi');
describe('App', () => {
  it('should render the search box to search videos',() => {
    const { getByTestId } = render(<App/>)

    expect(getByTestId('search-box-input')).toBeDefined();
  })

  it('should make an youtube api call to fetch videos for the search text',() => {
    jest.spyOn(youtubeApi, 'get').mockResolvedValue([]);
    const { getByTestId } = render(<App/>)

    const searchInput = getByTestId('search-box-input');
    fireEvent.change(searchInput,{target: {value: 'sampleText'}});
    fireEvent.submit(searchInput);

    expect(youtubeApi.get).toHaveBeenCalledTimes(1);
    expect(youtubeApi.get).toHaveBeenCalledWith("./search", {params: {q: "sampleText"}});
  })

  it('should render cat related videos while searching for cat',async() => {
    const mockVideos = {
      data: {
        items: [
          {
            id: {
              kind: "youtube#video",
              videoId: "123"
            },
            snippet: {
              title: "persian cat",
              description: "description about the persian",
              thumbnails: {
                medium : {
                  url: "https://i.ytimg.com/vi/AETFvQonfV8/default.jpg",
                }
              }
            }
          },
          {
            id: {
              kind: "youtube#video2",
              videoId: "456"
            },
            snippet: {
              title: "bengal cat",
              description: "description about the bengal cat",
              thumbnails: {
                medium : {
                  url: "https://i.ytimg.com/vi/AETFvQoabc/default.jpg",
                }
              }
            }
          }
        ] 
      }
    }
    jest.spyOn(youtubeApi, 'get').mockResolvedValue(mockVideos);
    const { container, getAllByTestId, getByTestId, rerender } = render(<App/>)

    const searchInput = getByTestId('search-box-input');
    fireEvent.change(searchInput,{target: {value: 'cat'}});
    fireEvent.submit(searchInput);
    await act(() => {
      rerender(<App/>);
    })
    expect(container).toMatchSnapshot();
    expect(getAllByTestId('video-item')).toHaveLength(2);
    expect(getAllByTestId('title')[0].textContent).toBe('persian cat');
    expect(getAllByTestId('title')[1].textContent).toBe('bengal cat');
  });
})
