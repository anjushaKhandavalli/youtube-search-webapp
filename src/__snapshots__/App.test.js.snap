// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render cat related videos while searching for cat 1`] = `
<div>
  <div
    class="sc-bdfBwQ dQmyfS"
  >
    <div
      class="sc-gsTCUz envuHy"
    >
      <form
        class="sc-dlfnbm bAyFQJ"
      >
        <div>
          <label
            class="sc-hKgILt ctfbPn"
            data-testid="search-box-label"
            for="search-box"
          >
            Search for videos via keywords:
          </label>
          <br />
          <div
            class="sc-gKsewC knLZKI"
          >
            <input
              class="sc-eCssSg eJznzd"
              data-testid="search-box-input"
              name="search-box"
              placeholder="Ex: dogs,cars etc.."
              type="text"
              value="cat"
            />
            <button
              class="sc-jSgupP bwGSVv"
            >
              Search
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="sc-iqHYGH ebMglc"
      data-testid="video-list"
    >
      <div
        class="infinite-scroll-component__outerdiv"
      >
        <div
          class="infinite-scroll-component "
          style="height: auto; overflow: auto;"
        >
          <div
            class="sc-iBPRYJ dkkhND"
            data-testid="video-item"
          >
            <img
              alt="description about the persian"
              class="sc-fubCfw faIHZd"
              data-testid="thumbnail"
              src="https://i.ytimg.com/vi/AETFvQonfV8/default.jpg"
            />
            <div
              class="sc-pFZIQ SbnXi"
            >
              <h2
                class="sc-jrAGrp fjjYoG"
                data-testid="title"
              >
                persian cat
              </h2>
              <h4
                class="sc-kEjbxe irossh"
                data-testid="description"
              >
                description about the persian
              </h4>
            </div>
          </div>
          <div
            class="sc-iBPRYJ dkkhND"
            data-testid="video-item"
          >
            <img
              alt="description about the bengal cat"
              class="sc-fubCfw faIHZd"
              data-testid="thumbnail"
              src="https://i.ytimg.com/vi/AETFvQoabc/default.jpg"
            />
            <div
              class="sc-pFZIQ SbnXi"
            >
              <h2
                class="sc-jrAGrp fjjYoG"
                data-testid="title"
              >
                bengal cat
              </h2>
              <h4
                class="sc-kEjbxe irossh"
                data-testid="description"
              >
                description about the bengal cat
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
